sys-apps/diskus/diskus-0.4.0.recipe

```diff
@@ -0,0 +1,190 @@
+CHECKSUM_SHA256_33="e464cd887e869cddcae8792a4ee31d23c7edd516700695608f5b98c67ee0131c"
+
+SOURCE_URI_34="https://static.crates.io/crates/smallvec/smallvec-0.6.5.crate#noarchive"
+CHECKSUM_SHA256_34="153ffa32fd170e9944f7e0838edf824a754ec4c1fc64746fcc9fe1f8fa602e5d"
+
+SOURCE_URI_35="https://static.crates.io/crates/textwrap/textwrap-0.10.0.crate#noarchive"
+CHECKSUM_SHA256_35="307686869c93e71f94da64286f9a9524c0f308a9e1c87a583de8e9c9039ad3f6"
+
+SOURCE_URI_36="https://static.crates.io/crates/lock_api/lock_api-0.1.4.crate#noarchive"
+CHECKSUM_SHA256_36="775751a3e69bde4df9b38dd00a1b5d6ac13791e4223d4a0506577f0dd27cfb7a"
+
+SOURCE_URI_37="https://static.crates.io/crates/parking_lot_core/parking_lot_core-0.3.1.crate#noarchive"
+CHECKSUM_SHA256_37="ad7f7e6ebdc79edff6fdcb87a55b620174f7a989e3eb31b65231f4af57f00b8c"
+
+SOURCE_URI_38="https://static.crates.io/crates/clap/clap-2.32.0.crate#noarchive"
+CHECKSUM_SHA256_38="b957d88f4b6a63b9d70d5f454ac8011819c6efa7727858f458ab71c756ce2d3e"
+
+SOURCE_URI_39="https://static.crates.io/crates/crossbeam-epoch/crossbeam-epoch-0.3.1.crate#noarchive"
+CHECKSUM_SHA256_39="927121f5407de9956180ff5e936fe3cf4324279280001cd56b669d28ee7e9150"
+
+SOURCE_URI_40="https://static.crates.io/crates/crossbeam-epoch/crossbeam-epoch-0.6.0.crate#noarchive"
+CHECKSUM_SHA256_40="9c90f1474584f38e270b5b613e898c8c328aa4f3dea85e0a27ac2e642f009416"
+
+SOURCE_URI_41="https://static.crates.io/crates/crossbeam-deque/crossbeam-deque-0.2.0.crate#noarchive"
+CHECKSUM_SHA256_41="f739f8c5363aca78cfb059edf753d8f0d36908c348f3d8d1503f03d8b75d9cf3"
+
+SOURCE_URI_42="https://static.crates.io/crates/parking_lot/parking_lot-0.6.4.crate#noarchive"
+CHECKSUM_SHA256_42="f0802bff09003b291ba756dc7e79313e51cc31667e94afbe847def490424cde5"
+
+SOURCE_URI_43="https://static.crates.io/crates/crossbeam-channel/crossbeam-channel-0.3.0.crate#noarchive"
+CHECKSUM_SHA256_43="548dfc2fa4933009668d4a51f9ef074b8af8f2d6a6410225d76d6b1c3c56ca6f"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+SECONDARY_ARCHITECTURES="x86"
+fi
+
+PROVIDES="
+	diskus$secondaryArchSuffix = $portVersion
+	cmd:diskus
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:cargo$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage diskus$secondaryArchSuffix \
+	$binDir/diskus
+
+BUILD()
+{
+	mkdir -p ~/.cargo/registry/cache/github.com-1ecc6299db9ec823
+	for i in {2..43}; do
+		eval temp=\$sourceDir$i
+		cp "$temp"/* ~/.cargo/registry/cache/github.com-1ecc6299db9ec823
+	done
+	cargo build --release
+}
+
+INSTALL()
+{
+	install -Dm555 -t $binDir target/release/diskus
+	install -Dm444 -t $docDir README.md
+}
+
+TEST()
+{
+	mkdir -p ~/.cargo/registry/cache/github.com-1ecc6299db9ec823
+	for i in {2..43}; do
+		eval temp=\$sourceDir$i
+		cp "$temp"/* ~/.cargo/registry/cache/github.com-1ecc6299db9ec823
+	done
```

Comment:
Since you'd have to build first before haikuporter let you run `--test`, maybe this is not needed?

Reply:
`/boot` is one of the temporary directories `haikuporter` makes for a proper environment, so it will be cleaned up/recreated before the testing.

Reply:
I believe `CARGO_HOME` can be set to elsewhere to prevent this.

Reply:
Done; thanks!